front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java

```diff
@@ -499,15 +510,24 @@ private void purgeOldVersions(String path) throws Exception {
 
   @Override
   public long getLastModified(ObjectType objectType) {
+      return getLastModifiedHelper(objectType, true);
+  }
+
+  private long getLastModifiedHelper(ObjectType objectType, boolean expected) {
       String path = daoRoot(objectType.group) + '/' + LAST_MODIFIED_FILENAME;
       try {
           return obj_api.get(bucketName, path).execute().getUpdated().getValue();
       } catch (HttpResponseException e) {
           if (e.getStatusCode() == 404) {
-              log.error("No timestamp file at {}. Creating a new one.", path);
-              writeLastModified(objectType.group);
+              if (expected) {
+                  log.error("No timestamp file at {}. Creating a new one.", path);
+              } else {
+                  log.info("No timestamp file at {}. Creating a new one.", path);
+              }
+              writeLastModified(objectType.group, expected);
               return 0L;
```

Comment:
Not sure what the `0L` response is meant to indicate ... would seem like it should either be the last modified time if a file was written ... or an exception if the write failed?